src/utils/events.js

```diff
@@ -0,0 +1,323 @@
+import client from '../api/client'
+import moment from 'moment'
+import constants from '../constants'
+import {get, isUndefined, set} from 'lodash'
+
+const {PUBLICATION_STATUS, EVENT_STATUS} = constants
+
+export class EventQueryParams {
+    admin_user = null
+    end = null
+    include = null
+    nocache = null
+    page = null
+    page_size = null
+    publication_status = null
+    publisher = null
+    show_all = null
+    sort = null
+    start = null
+    super_event = null
+    text = null
+
+    setPublisher (publisher) {
+        this.publisher = publisher && publisher.join()
+    }
+    setSort(sortBy, sortDirection) {
+        this.sort = sortDirection === 'desc' ? `-${sortBy}` : sortBy
+    }
+    setText(query) {
+        this.text = encodeURI(query)
+    }
+    get values() {
+        return Object.keys(this)
+            .filter(key => typeof this[key] !== 'function')
+            .reduce((acc, key) => set(acc, key, this[key]), {})
+    }
+}
+
+/**
+ * Fetches event data based on given ID
+ * @param eventId       ID of event to fetch
+ * @param queryParams   EventQueryParams class object containing the query parameters for the request
+ * @param fetchSuper    Whether super event data should be fetched for the given event
+ * @returns {Promise<*>}    Returns a promise containing event, sub event and possibly super event data depending on given params
+ */
+export const fetchEvent = (eventId, queryParams, fetchSuper = false) =>
+    new Promise(async (resolve) => {
```

Comment:
If you turn `fetchEvent` into an `async` function you can remove the `Promise` entirely since an `async` function always returns a promise.